### PR TITLE
Fix for application search UX issue

### PR DESCRIPTION
**Pull request by kayathiri4**
### Purpose
User have to get the results even they search with something like `name co queryString` with the default search in console and my-account
Advanced search is disabled for myaccount by default in identity apps and asgardeo

**Console**

1. Existing
Screenshot from 2021-06-23 15-54-56

2. Modified 
Screenshot from 2021-06-23 15-47-16

**My account**

1. Existing
Screenshot from 2021-06-23 15-36-30

2. Modified
Screenshot from 2021-06-23 15-49-45

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- https://github.com/wso2-enterprise/asgardeo-app-extensions/pull/797

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/myaccount/src/components/shared/advanced-search-with-basic-filters.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ import React, { FunctionComponent, ReactElement, useState } from "react";
 import { useTranslation } from "react-i18next";
 import { Divider, Form, Grid } from "semantic-ui-react";
 import { getAdvancedSearchIcons } from "../../configs";
+import {commonConfig} from "../../extensions";
```
Comment (NipuniBhagya) [inline apps/myaccount/src/components/shared/advanced-search-with-basic-filters.tsx:27]:
```suggestion
import { commonConfig } from "../../extensions";
```

**Review by NipuniBhagya — APPROVED** [review]

File: apps/console/src/features/core/components/advanced-search-with-basic-filters.tsx [hunk 2]
```diff
@@ -241,6 +241,8 @@ export const AdvancedSearchWithBasicFilters: FunctionComponent<AdvancedSearchWit
             submitted={ isFormSubmitted }
             dropdownPosition={ dropdownPosition }
             triggerClearQuery={ triggerClearQuery }
+            defaultFilterConditionOptions = { defaultFilterConditionOptions }
```
Comment (brionmario) [inline apps/console/src/features/core/components/advanced-search-with-basic-filters.tsx:244]: The attribute should be `filterConditionOptions` here. If `filterConditionOptions` is not overriden from outside, we should use the `defaultFilterConditionOptions`. 

```suggestion
            filterConditionOptions = { filterConditionOptions || defaultFilterConditionOptions }
```

File: apps/console/src/features/core/components/advanced-search-with-basic-filters.tsx [hunk 3]
```diff
@@ -241,6 +241,8 @@ export const AdvancedSearchWithBasicFilters: FunctionComponent<AdvancedSearchWit
             submitted={ isFormSubmitted }
             dropdownPosition={ dropdownPosition }
             triggerClearQuery={ triggerClearQuery }
+            filterConditionOptions = { filterConditionOptions || defaultFilterConditionOptions }
```
Comment (brionmario) [inline apps/console/src/features/core/components/advanced-search-with-basic-filters.tsx:244]:
```suggestion
            filterConditionOptions={ filterConditionOptions || defaultFilterConditionOptions }
```

File: apps/console/src/features/core/components/advanced-search-with-basic-filters.tsx [hunk 4]
```diff
@@ -241,6 +241,8 @@ export const AdvancedSearchWithBasicFilters: FunctionComponent<AdvancedSearchWit
...
+            filterAttributeOptions = { filterAttributeOptions }
```
Comment (brionmario) [inline apps/console/src/features/core/components/advanced-search-with-basic-filters.tsx:245]:
```suggestion
            filterAttributeOptions={ filterAttributeOptions }
```

File: apps/myaccount/src/components/shared/advanced-search-with-basic-filters.tsx [hunk 5]
```diff
@@ -221,22 +227,25 @@ export const AdvancedSearchWithBasicFilters: FunctionComponent<AdvancedSearchWit
...
             placeholder={ placeholder }
             resetSubmittedState={ handleResetSubmittedState }
             searchOptionsHeader={ t("myAccount:components.advancedSearch.options.header") }
+            enableQuerySearch={ enableQuerySearch }
             externalSearchQuery={ externalSearchQuery }
             submitted={ isFormSubmitted }
             dropdownPosition={ dropdownPosition }
             triggerClearQuery={ triggerClearQuery }
             data-testid={ testId }
+            filterConditionOptions = { filterConditionOptions || defaultFilterConditionOptions }
```
Comment (brionmario) [inline apps/myaccount/src/components/shared/advanced-search-with-basic-filters.tsx:247]:
```suggestion
            filterConditionOptions={ filterConditionOptions || defaultFilterConditionOptions }
```

File: modules/react-components/src/components/input/advanced-search.tsx [hunk 6]
```diff
@@ -267,7 +277,26 @@ export const AdvancedSearch: FunctionComponent<PropsWithChildren<AdvancedSearchP
             if (internalSearchQuery === "") {
                 query = null;
             } else {
-                query = `${ defaultSearchStrategy } ${ internalSearchQuery }`;
+                let advancedSearch = false;
```
Comment (brionmario) [inline modules/react-components/src/components/input/advanced-search.tsx:280]:
```suggestion
                let advancedSearch: boolean = false;
```

File: modules/react-components/src/components/input/advanced-search.tsx [hunk 7]
```diff
@@ -267,7 +277,26 @@ export const AdvancedSearch: FunctionComponent<PropsWithChildren<AdvancedSearchP
...
+                const terms = internalSearchQuery.split(" ");
```
Comment (brionmario) [inline modules/react-components/src/components/input/advanced-search.tsx:281]:
```suggestion
                const terms: string[] = internalSearchQuery.split(" ");
```

**Review by brionmario — APPROVED** [review]

File: modules/react-components/src/components/input/advanced-search.tsx [hunk 8]
```diff
@@ -120,6 +120,14 @@ export interface AdvancedSearchPropsInterface extends TestableComponentInterface
      * Enable query search with shift and enter.
      */
     enableQuerySearch?: boolean;
+    /**
+     * Default filter conditions..
```
Comment (nilasini) [inline modules/react-components/src/components/input/advanced-search.tsx:124]: remove extra .

Comment (kayathiri4) [reply modules/react-components/src/components/input/advanced-search.tsx:124]: Addressed with https://github.com/wso2/identity-apps/pull/2574

**Pull request merged**